flytekit/bin/entrypoint.py

```diff
@@ -137,7 +137,41 @@ def _dispatch_execute(
             logger.warning("Task produces no outputs")
             output_file_dict = {_constants.OUTPUT_FILE_NAME: _literal_models.LiteralMap(literals={})}
         elif isinstance(outputs, _literal_models.LiteralMap):
-            output_file_dict = {_constants.OUTPUT_FILE_NAME: outputs}
+            offloaded_literals: Dict[str, _literal_models.Literal] = {}
+            literal_map_copy = {}
+
+            min_offloaded_size = int(os.environ.get("FK_L_MIN_SIZE_MB", "10")) * 1024 * 1024
+            max_offloaded_size = int(os.environ.get("FK_L_MAX_SIZE_MB", "1000")) * 1024 * 1024
+
+            # Go over each output and create a separate offloaded in case its size is too large
+            for k, v in outputs.literals.items():
+                lit = v.to_flyte_idl()
+                if lit.ByteSize() >= max_offloaded_size:
+                    raise ValueError(
+                        f"Literal {k} is too large to be offloaded. Max literal size is {max_offloaded_size} whereas the literal size is {lit.ByteSize()} bytes"
+                    )
+
+                if lit.ByteSize() >= min_offloaded_size:
+                    logger.debug(f"Literal {k} is too large to be inlined, offloading to metadata bucket")
+
+                    # TODO: hash calculation
+
+                    offloaded_filename = f"{k}_offloaded_metadata.pb"
+
+                    offloaded_literal = _literal_models.Literal(
+                        offloaded_metadata=_literal_models.LiteralOffloadedMetadata(
+                            uri=f"{ctx.user_space_params.output_metadata_prefix}/{offloaded_filename}",
+                            size_bytes=lit.ByteSize(),
+                            # TODO: do I have to set the inferred literal type?
```

Comment:
We do need to add inferred type, since during consumption we validate the input type

https://github.com/unionai/flyte/blob/5e9ec71a3b55003955642aeeaaf6dbeb43b3a279/flytepropeller/pkg/compiler/validators/utils.go#L271-L272

https://github.com/unionai/flyte/blob/2c070c301ccb37c5a336dd7022a1d9846d9bb173/flyteadmin/pkg/manager/impl/validation/execution_validator.go#L100

Reply:
Wont we still need this in the new isInstance form https://github.com/flyteorg/flyte/pull/5909/files#diff-af9c7a5cd03e0e860c155861edcf0b96863e8781526f51d1b13c7f9278ffa72fR415

Reply:
Let's separate those two discussions. We might be able to remove `LiteralTypeForLiteral`, but that shouldn't block us from getting the actual literal type from the typed interface in flytekit and use it here.

Also, if/when we end up removing `LiteralTypeForLiteral` we should use a different field to store the offloaded literal literal type and not use `inferred` in the name since it'll be the exact type.